### PR TITLE
Add eslint-plugin-grommet and address issues

### LINKED ISSUE #3403: NextJS 13 + eslint-plugin-grommet compatibility
With NextJS 13, eslint-plugin-grommet rule definitions can't be found.

To do:
- [x] Investigate why and put up a PR so we can enable the plugin again

### LINKED ISSUE #3404: Improve rendering of MetricHPE
This ticket is intended to track work related to improving the rendering/alignment of MetricHPE. Reopening this work based on this slack thread: https://hpe.slack.com/archives/C04LMJWQT/p1686071250418569

The end goal of this work is to:
1. Have the font render as cleanly as possible across browsers and viewports.
2. Have the font render with the correct vertical positioning so it is visually centered with icons and other visual elements.
3. Have approval from both marketing and product teams on the direction (key stakeholders tbd)
4. Ideally, continue to consume the font from an HPE domain to have consistency of fonts across business units

Previous discussions and work can be found below:
- https://github.com/grommet/hpe-design-system/issues/2990
- https://github.com/grommet/hpe-design-system/issues/2080
- https://github.com/grommet/hpe-design-system/issues/2854
- https://www.figma.com/file/yEueZRtVp6AuofGfzwb055/MetricHPE-Font?type=whiteboard&node-id=0-1&t=xXXmJ7lkFn1ayFBu-0

TLDR from previous discussions:
- MetricHPE = official version of font supported by brand central
- MetricHPEXS = version of font created to try to solve some baseline issues in MetricHPE
- MetricHPEXS introduced legibility issues for i/j on certain browsers
- In order to resolve the legibility issues, grommet-theme-hpe reverted to use of MetricHPE
- Given this, some of the baseline misalignments with other elements like icons still exists and should be solved for

Avenues to explore:
1. Explore potential CSS solutions that use the current version of MetricHPE but tweak font-size, line-height, etc. combinations to achieve the desired presentation
2. Work with brand to contact a foundry that can produce an improved version of the font

Tasks to be done in the near term:
- ~~Add flag in grommet-theme-hpe that turns on grommet functionality of `text-rendering: auto` and `-webkit-font-smoothing: antialiased`~~ Removing, @samuelgoff shared this did not have improvement even at the sub-pixel level

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-INSERT_PR_#_HERE--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Adds `eslint-plugin-grommet` back to eslint config. Addressed the issues being flagged by the rules.

For some areas where the rules felt inaccurate or overly strict, I filed issues to the `eslint-plugin-grommet` repo:
- https://github.com/grommet/eslint-plugin-grommet/issues/46
- https://github.com/grommet/eslint-plugin-grommet/issues/47
- https://github.com/grommet/eslint-plugin-grommet/issues/48

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
Closes #3403

### CODE REVIEW

File: aries-site/src/examples/templates/forms/RequiredFieldsExample.js [hunk 1]
```diff
@@ -90,6 +92,8 @@ export const RequiredFieldsExample = () => {
           >
             <Select options={superPower} id="superPower" name="superPower" />
           </FormField>
+          {/* TO DO - enhance formfield-htmlfor-id to support Select */}
```
Comment (jcfilben) [inline aries-site/src/examples/templates/forms/RequiredFieldsExample.js:95]: Do we have an issue to track this?

**Review by jcfilben — APPROVED** [review]
Nice cleanup!

File: aries-site/src/examples/templates/forms/RequiredFieldsExample.js (hunk 1 above)
Comment (jcfilben) [reply aries-site/src/examples/templates/forms/RequiredFieldsExample.js:95]: Nevermind, I just found it https://github.com/grommet/eslint-plugin-grommet/issues/46

Comment (taysea) [reply aries-site/src/examples/templates/forms/RequiredFieldsExample.js:95]: Good catch -- let me adjust the comment

Comment (taysea) [inline aries-site/src/examples/templates/forms/RequiredFieldsExample.js:95]:
```suggestion
          {/* https://github.com/grommet/eslint-plugin-grommet/issues/46 */}
```

Comment (taysea) [reply aries-site/src/examples/templates/forms/RequiredFieldsExample.js]: Done.

**Review by halocline — APPROVED** [review]
